astrovascpy/bloodflow.py

```diff
@@ -654,15 +652,19 @@ def _solve_linear(laplacian, input_flow):
         if sparse.issparse(input_flow):
             input_flow = input_flow.toarray()
 
-        result = np.zeros(np.shape(laplacian)[0], dtype=laplacian.dtype)
+        n_nodes = np.shape(laplacian)[0]
+        result = np.zeros(shape=n_nodes, dtype=laplacian.dtype)
     else:
         result = None
 
     # in os.getenv() the second argument refers to the default value
-    if (
-        bool(int(os.getenv("DEBUG_BFS", "0")))
-        or os.getenv("BACKEND_SOLVER_BFS", "petsc") == "scipy"
-    ):
+    if os.getenv("BACKEND_SOLVER_BFS", "scipy") == "scipy":
+        if MPI_RANK == 0 and n_nodes > 2e6:
+            L.warning(
+                "The program can be slow and the result not very accurate."
```

Comment:
I would add 2 information to the warning messages:
- the fact that the petsc solver is recommended for graph with more than 2e6 nodes
- how to enable the petsc solver

Reply:
Done.  Now there is only one warning message with the instruction to select the solver. Please let me know if it is ok.